DoSomething/gladiator
# DB Restructure

#### What's this PR do?

Updates the DB structure in Gladiator as outlined in #45 

* Removed old migrations, and made new ones. 
* Added a new `Contests` table and a model that defined it's relationship to waiting rooms and competitions
* Renamed pivot tables so they get run last. 
* Added a new `ContestTableSeeder` that makes 3 contests.
* Added a `ContestServiceProvider` that listens for the `created` event on Contest models and automatically creates a new waiting room for the competition. 
* Created a competition seeder that just creates three empty competitions for a Contest.

NOTES/QUESTIONS -

* I wasn't sure about the `duration` field. My thought is we can make the decision to always store this as an `integer` representing days and then on the front end we can provide a slicker interface with like units (days, weeks, months, etc) that gets converted into days before it gets stored in the db. But I am open to doing something different. 

* Because of the ^^ open question I didn't implement any fancy calculation of competition start and end date yet. 

* I didn't touch any CRUD stuff or other logic to keep this as small as I could, so we can fix those up in other issues. 

#### How should this be manually tested?

Pull down this branch
Run `composer dump-autoload`
Run `php artisan:refresh` to get new tables
Run `php artisan db:seed` to seed the DB

Review the DB to make sure all the correct columns are there and the relationships look right.

#### What are the relevant tickets?

Fixes #45 

@angaither @deadlybutter @weerd
